.clang-format

```diff
@@ -1,8 +1,37 @@
+
+# Based on the https://www.kernel.org/doc/Documentation/CodingStyle
```

Comment:
Thanks for adding the comments explaining the options. This link seems to be broken, though. Plus our style isn't really derived from the Linux kernel style at all, is it?

Reply:
Sorry, did that in a haste yesterday, not sure where that one came from. Updated to link to LLVM.